Client/mods/deathmatch/logic/lua/CLuaFunctionDefs.World.cpp

```diff
@@ -1764,6 +1764,263 @@ int CLuaFunctionDefs::GetBirdsEnabled(lua_State* luaVM)
+    {
+        if (iWeatherID >= 0 && iWeatherID <= 23 && iCycleNum >= 0 && iCycleNum <= 7)
+        {
+            if (!strPropertyName.empty())
+            {
+                float fValue = g_pGame->GetTimeCycle()->GetTimeCycle(iWeatherID, iCycleNum, strPropertyName);
+                if (fValue != -1)
+                    lua_pushnumber(luaVM, fValue);
+                else
+                    argStream.SetCustomError("Property doesn't exist");
+            }
+            else
+            {
+                std::map<const char*, float> m_Properties;
+                g_pGame->GetTimeCycle()->GetTimeCycle(iWeatherID, iCycleNum, m_Properties);
+                if (!m_Properties.empty())
+                {
+                    lua_newtable(luaVM);
+                    std::map<const char*, float>::iterator it = m_Properties.begin();
+                    for (; it != m_Properties.end(); it++)
+                    {
+                        lua_pushstring(luaVM, it->first);
+                        lua_pushnumber(luaVM, it->second);
+                        lua_settable(luaVM, -3);
+                    }
+                }
+            }
+        }
+        else if (iWeatherID >= 0 && iWeatherID <= 23 && iCycleNum == -1)
+        {
+            std::map<int, std::map<const char*, float>> m_Properties;
+            g_pGame->GetTimeCycle()->GetTimeCycle(iWeatherID, m_Properties);
+            if (!m_Properties.empty())
+            {
+                lua_newtable(luaVM);
+                std::map<int, std::map<const char*, float>>::iterator it = m_Properties.begin();
+                for (; it != m_Properties.end(); it++)
+                {
+                    lua_newtable(luaVM);
+                    lua_pushnumber(luaVM, it->first);
+                    lua_pushvalue(luaVM, -2);
+                    lua_settable(luaVM, -4);
+
+                    std::map<const char*, float>::iterator it2 = it->second.begin();
+                    for (; it2 != it->second.end(); it2++)
+                    {
+                        lua_pushstring(luaVM, it2->first);
+                        lua_pushnumber(luaVM, it2->second);
+                        lua_settable(luaVM, -3);
+                    }
+                    lua_pop(luaVM, 1);
+                }
+            }
+        }
+        else
+            lua_pushboolean(luaVM, false);
+        return 1;
+    }
+    else
+        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+
+    lua_pushboolean(luaVM, false);
+    return 1;
+}
+
+int CLuaFunctionDefs::GetOriginalTimeCycle(lua_State* luaVM)
+{
+    //  mixed getOriginalTimeCycle ( int weatherID, [int cycle = -1, string property = ""] )
+
+    int              iWeatherID, iCycleNum;
+    SString          strPropertyName;
+    CScriptArgReader argStream(luaVM);
+
+    argStream.ReadNumber(iWeatherID);
+    argStream.ReadNumber(iCycleNum, -1);
+    argStream.ReadString(strPropertyName, "");
+
+    if (!argStream.HasErrors())
```

Comment:
These errors should be changed accordingly (https://github.com/multitheftauto/mtasa-blue/issues/821#issuecomment-522187565).